File: hw/tipan/u3/hw28Strategy/StrategyJava/Strategy/src/main/java/ec/edu/espe/strategy/model/BubbleSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy.model;

/**
 *
 * @author ronal
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {

        int[] array = data.clone();

        for (int i = 0; i < array.length - 1; i++) {
            for (int j = 0; j < array.length - i - 1; j++) {

                if (array[j] > array[j + 1]) {

                    int aux = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = aux;

                }

            }
        }

        return array;

    }

}

File: hw/tipan/u3/hw28Strategy/StrategyPython/model/bubble_sort.py
from model.sorting_strategy import SortingStrategy


class BubbleSort(SortingStrategy):

    def sort(self, data):

        array = data.copy()

        for i in range(len(array) - 1):
            for j in range(len(array) - i - 1):
                if array[j] > array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]

        return array